p2p/host/eventbus/basic.go

```diff
@@ -57,12 +67,22 @@ func NewBus() event.Bus {
 	}
 }
 
+func NewBusWithMetrics() event.Bus {
```

Comment:
This works totally fine, but we usually use the option pattern:
```go
type Option func(*basicBus) error

func NewBus(opts ...Option) event.Bus
```

Reply:
will fix. This would also be backwards compatible. nice :)

Reply:
This is what we did for the swarm: https://github.com/libp2p/go-libp2p/blob/master/p2p/net/swarm/swarm.go#L74-L79

Reply:
fixed. Not returning error for now since there's only one option that can't error.